reference/customers.v3.yml

```diff
@@ -697,6 +704,114 @@ paths:
+        required: true
+      responses:
+        "200":
+          description: Customer settings is returned.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/responses/CustomerSettingsResponse"
+              examples:
+                data:
+                  value:
+                    privacy_settings:
+                      ask_shopper_for_tracking_consent: true
+                      policy_url: "https://bigcommmerce.com/policy"
+                    customer_group_settings:
+                      guest_customer_group_id: 1
+                      default_customer_group_id: 1
+  /customers/settings/channels/{channel_id}:
+    get:
+      tags:
+        - Retrieving Customer settings per channel
+      description: Gets the customer settings per channel.
+      parameters:
+        - in: path
+          name: channel_id
+          schema:
+            type: integer
+          required: true
+      responses:
+        "200":
+          description: Customer settings for this channel is returned.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/responses/CustomerChannelSettingsResponse"
+              examples:
+                data:
+                  value:
+                    privacy_settings:
+                      ask_shopper_for_tracking_consent: true
+                      policy_url: "https://bigcommmerce.com/policy"
+                    customer_group_settings:
+                      guest_customer_group_id: 1
+                      default_customer_group_id: 1
+                    allow_global_logins: true
+    put:
+      tags:
+        - Declaring Customer settings per channel
+      description: Update the customer settings per channel.
+      parameters:
+        - in: path
+          name: channel_id
+          schema:
+            type: integer
+          required: true
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CustomerChannelSettingsObject"
+        required: true
+      responses:
+        "200":
+          description: Customer settings is returned.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/responses/CustomerChannelSettingsResponse"
+              examples:
+                data:
+                  value:
+                    privacy_settings:
+                      ask_shopper_for_tracking_consent: true
+                      policy_url: "https://bigcommmerce.com/policy"
+                    customer_group_settings:
+                      guest_customer_group_id: 1
+                      default_customer_group_id: 1
+                    allow_global_logins: true
   /customers/attributes:
     get:
       description: Returns a list of Customer Attributes. Optional filter parameters can be passed in.
```

Comment:
Returns a list of Customer Attributes. You can pass in optional filter parameters.